orchestra/actions/install.py

```diff
@@ -31,38 +39,19 @@ def _run(self, args):
 
         pre_file_list = self._index_directory(tmp_root + orchestra_root, relative_to=tmp_root + orchestra_root)
 
-        start_time = time.time()
-
-        if self.from_binary_archives and self._binary_archive_exists():
-            logger.info("Fetching binary archive")
-            self._fetch_binary_archive()
-            logger.info("Extracting binary archive")
-            self._install_from_binary_archives()
-            source = "binary archive"
-
-            logger.info("Removing conflicting files")
-            self._remove_conflicting_files()
-            self.update_binary_archive_symlink()
-        elif not self.from_binary_archives or self.fallback_to_build:
-            self._install(args.quiet, args.test)
-            source = "build"
-
-            logger.info("Removing conflicting files")
-            self._remove_conflicting_files()
-
-            if self.build.component.skip_post_install:
-                logger.info("Skipping post install")
-            else:
-                self._post_install(args.quiet)
-
-            if args.create_binary_archives:
+        install_start_time = time.time()
+        if self.allow_binary_archive and self._binary_archive_exists():
+            self._install_from_binary_archive(args)
+            source = "binary archives"
+        elif self.allow_build:
+            self._build_and_install(args)
+            if getattr(args, "create_binary_archives", False):
```

Comment:
This is not OK, this need to become an argument

Reply:
You're right, fixed in PR #9